### PR TITLE
Ywmei/handle annotation during decoding

### PR DESCRIPTION
**Pull request by ywmei-brt1**

### CODE REVIEW

File: firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt [hunk 1]
```diff
@@ -15,6 +15,10 @@
 package com.google.firebase.firestore.ktx.serialization
 
 import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.ThrowOnExtraProperties
+import com.google.firebase.firestore.encoding.FirestoreAbstractDecoder as FireDecoder
```
Comment (rlazo) [inline firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt:19]: I'd rather not use this feature.

File: firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt [hunk 2]
```diff
@@ -143,6 +152,34 @@ private abstract class FirestoreAbstractDecoder(
...
 
+/** replaces the @[DocumentId] annotated field with [DocumentReference] value. */
+private fun Map<String, Any?>.replaceKDocumentIdFieldWithCurrentDocRef(
+    descriptor: SerialDescriptor,
+    docRef: DocumentReference
+): MutableMap<String, Any?> =
+    this.toMutableMap().apply {
+        for (propertyName in descriptor.elementNames) {
+            val propertyIndex: Int = descriptor.getElementIndex(propertyName)
+            val annotationsOnProperty: List<Annotation> =
```
Comment (rlazo) [inline firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt:163]: should be ok to not include types.

**Review by rlazo — APPROVED** [review]

File: firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt (hunk 2 above)
Comment (ywmei-brt1) [reply firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt:163]: thank you.

File: firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt (hunk 1 above)
Comment (ywmei-brt1) [reply firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/FirestoreMapDecoder.kt]: I have removed this feature by changed the Kotlin's abstract decoder class's name.

**Pull request merged**